src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java

```diff
@@ -162,23 +151,20 @@ protected void configure(HttpSecurity http) throws Exception {
             .sessionManagement()
             .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
         .and()
+            // api
             .authorizeRequests()
-            .antMatchers("/api/register").permitAll()
-            .antMatchers("/api/activate").permitAll()
-            .antMatchers("/api/authenticate").permitAll()
-            .antMatchers("/api/account/reset-password/init").permitAll()
-            .antMatchers("/api/account/reset-password/finish").permitAll()
-            .antMatchers("/api/lti/launch/*").permitAll()
-            .antMatchers("/api/lti13/auth-callback").permitAll()
-            .antMatchers("/api/programming-exercises/*/build-plan").permitAll()
-            .antMatchers("/api/**").authenticated()
+            .antMatchers("/api/admin/**").hasAuthority(Role.ADMIN.getAuthority())
+            .antMatchers("/api/public/**").permitAll()
+            // TODO: Remove the following three lines in January 2024 together with LegacyResource
```

Comment:
```suggestion
            // TODO: Remove the following three lines in next major release together with LegacyResource
```

Reply:
Stephan wanted to leave it for a year, not for the next major release so we stick with that for now. However, I can adjust it to February now lol